src/model/transaction/Transaction.ts

```diff
@@ -156,4 +156,18 @@ export abstract class Transaction {
         const versionDTO = this.networkType.toString(16) + '0' + this.version.toString(16);
         return parseInt(versionDTO, 16);
     }
+
+    /**
+     * @description re-aplly a given value to the transaction in an immutable way
+     * @param {Deadline} deadline
+     * @returns {Transaction}
+     * @memberof Transaction
+     */
+    public reapplygiven(deadline: Deadline): Transaction {
+        if (this.isUnannounced()) {
+            return Object.assign({__proto__: Object.getPrototypeOf(this)}, this, {deadline});
+        } else {
+            throw new Error('an Announced transaction can\'t be modified');
```

Comment:
does exist a case where you want to reply an already announced transaction?

I would do:
```typescript
if (this.isUnannounced()) {
    return Object.assign({__proto__: Object.getPrototypeOf(this)}, this, {deadline});
}
throw new Error('an Announced transaction can\'t be modified');
```

The code is more clean, personal style :smile: